programs/CMakeLists.txt

```diff
@@ -30,49 +30,42 @@ endif()
 configure_file(${PROJECT_SOURCE_DIR}/programs/config.h.in 
                ${PROJECT_SOURCE_DIR}/programs/config.h)
 
-set(TEST_PROGRAMS_DEFINITIONS _POSIX_C_SOURCE=200809L
-                              _FILE_OFFSET_BITS=64
-                              HAVE_CONFIG_H)
-
-if (DEFLATE_TEST_SUPPORT__DO_NOT_USE)
-  list(APPEND TEST_PROGRAMS_DEFINITIONS TEST_SUPPORT__DO_NOT_USE)
-endif()
-
-if (BUILD_SHARED_LIBS)
-  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-endif()
-
 # Small library for common code
-add_library(programs)
-target_sources(programs PRIVATE prog_util.c test_util.c tgetopt.c)
-target_compile_definitions(programs PRIVATE ${TEST_PROGRAMS_DEFINITIONS})
-target_link_libraries(programs deflate)
+add_library(libdeflate_programs_common STATIC prog_util.c tgetopt.c)
+target_compile_definitions(libdeflate_programs_common PUBLIC _POSIX_C_SOURCE=200809L HAVE_CONFIG_H _FILE_OFFSET_BITS=64)
+target_link_libraries(libdeflate_programs_common PUBLIC libdeflate)
 
 # cli
-set(PROGRAM_TARGET_NAMES
-  gzip
-  benchmark
-  checksum
-)
-foreach(PROGRAM_TARGET_NAME ${PROGRAM_TARGET_NAMES})
-  add_executable(${PROGRAM_TARGET_NAME} ${PROGRAM_TARGET_NAME}.c)
-  target_link_libraries(${PROGRAM_TARGET_NAME} programs deflate z)
-  target_compile_definitions(${PROGRAM_TARGET_NAME} PRIVATE ${TEST_PROGRAMS_DEFINITIONS})
-endforeach()
+add_executable(libdeflate-gzip gzip.c)
+target_link_libraries(libdeflate-gzip PRIVATE libdeflate_programs_common)
+
+install(TARGETS libdeflate-gzip DESTINATION ${CMAKE_INSTALL_BINDIR})
 
```

Comment:
```
rm -rf build ; mkdir build ; cd build ; cmake -DDEFLATE_BUILD_TEST_PROGRAMS=OFF .. ; make ; cd ..
```

I tried to build my gist that way, but I still get those small packages and library built. Am I doing something wrong ?

Reply:
libuv has those cmake_dependent_options, maybe we could use this ?

https://github.com/libuv/libuv/blob/v1.x/CMakeLists.txt#L20

Reply:
You mean libdeflate-gzip? It's intended, it's not a test, it's even installed in the Makefile.